components/brave_welcome_ui/state/data-context-provider.tsx

```diff
@@ -8,53 +8,73 @@ import { addWebUiListener } from 'chrome://resources/js/cr.js'
 
 import DataContext from '../state/context'
 import { ViewType, Scenes } from '../state/component_types'
-import { useInitializeImportData, useProfileCount } from '../state/hooks'
+import {
+  useInitializeImportData,
+  useProfileCount,
+  useViewTypeTransition
+} from '../state/hooks'
+import { loadTimeData } from '$web-common/loadTimeData'
 
 interface DataContextProviderProps {
   children: React.ReactNode
 }
 
 function DataContextProvider (props: DataContextProviderProps) {
-  const [viewType, setViewType] = React.useState<ViewType | undefined>(undefined)
-  const [currentSelectedBrowser, setCurrentSelectedBrowser] = React.useState<string | undefined>(undefined)
+  const [viewType, setViewType] = React.useState<ViewType | undefined>(
+    undefined
+  )
+  const [currentSelectedBrowser, setCurrentSelectedBrowser] = React.useState<
+    string | undefined
+  >(undefined)
   const { browserProfiles } = useInitializeImportData()
   const { profileCountRef, incrementCount, decrementCount } = useProfileCount()
   const [scenes, setScenes] = React.useState<Scenes | undefined>(undefined)
 
+  const currentSelectedBrowserProfiles = React.useMemo(() => {
+    return browserProfiles?.filter(
+      (profile) => profile.browserType === currentSelectedBrowser
+    )
+  }, [browserProfiles, currentSelectedBrowser])
+
   const store = {
```

Comment:
is the type for `store` even being inferred here? if not we can make the `interface Store {}` exportable and explicitly anotate the type here.

Reply:
Yeah, that's a good idea. Done - used `Store` interface we had in ../state/context.ts